osu.Game/Rulesets/Mods/IMod.cs

```diff
@@ -81,5 +83,29 @@ public interface IMod : IEquatable<IMod>
         /// Create a fresh <see cref="Mod"/> instance based on this mod.
         /// </summary>
         Mod CreateInstance() => (Mod)Activator.CreateInstance(GetType())!;
+
+        /// <summary>
+        /// Whether any user adjustable setting attached to this mod has a non-default value.
+        /// </summary>
+        bool HasNonDefaultSettings
```

Comment:
I'd hope it's obvious to anyone that this is the *instantaneous* state and you need a setting tracker for this to work reliably in a context where mods are mutable, but maybe worth adding xmldoc to spell that out?